packages/button/README.md

```diff
@@ -91,7 +91,9 @@ npm install @leafygreen-ui/button
 | `as`           | `'a' \| 'button'`                                                                          | Determines the root element. An `a` tags can be supplied to replace `button` from being the DOM element that wraps the component.                                                                               | `button`    |
 | `href`         | `string`                                                                                   | If a href is supplied it will change the `as` value, such that the component renders inside of an `a` tag instead of inside of a `button` tag.                                                                  |             |
 | `leftGlyph`    | `React.ReactElement`                                                                       | Glyph that will appear to the left of text, if there is text provided via the children prop. If no children are supplied to the component, passing an Icon here will render the button as an icon-only button.  |             |
-| `rightGlyph`   | `React.ReactElement`                                                                       | Glyph that will appear to the right of text, if there is text provided via the children prop. If no children are supplied to the component, passing an Icon here will render the button as an icon-only button. |             |
+| `rightGlyph`   | `React.ReactElement`                                                                       | Glyph that will appear to the right of text, if there is text provided via the children prop. If no children are supplied to the component, passing an Icon here will render the button as an icon-only button. |
+| `isLoading`    | `boolean`                                                                                  | Indicates whether the Button is in a loading state                                                                                                                                                              | `false`     |
+| `loadingText`  | `string`                                                                                   | String displayed in place of `children` while the button is in a loading state                                                                                                                                  |             |
```

Comment:
i don't support this API. why not pass through children

Reply:
When there is no loading text, we want the button to maintain the width of its children when `isLoading` is true. This means we need some distinction between loadingText and the normally displayed text. We also don't want the user to have `isLoading ? 'some loading text' : 'normal text'` logic in their implementation. I figured this would be a cleaner DX.